cloudstack/data_source_cloudstack_template.go

```diff
@@ -0,0 +1,144 @@
+package cloudstack
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/xanzy/go-cloudstack/cloudstack"
+	"log"
+	"regexp"
+	"time"
+)
+
+func dataSourceCloudstackTemplate() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCloudstackTemplateRead,
+		Schema: map[string]*schema.Schema{
+			"filter": dataSourceFiltersSchema(),
+			"templatefilter": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			// Computed values
+			"template_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"account": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"displaytext": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"format": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hypervisor": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"size": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func dataSourceCloudstackTemplateRead(d *schema.ResourceData, meta interface{}) error {
+	cs := meta.(*cloudstack.CloudStackClient)
+	ltp := cloudstack.ListTemplatesParams{}
```

Comment:
_nitting_ In other places within this provider we just call the param struct `p` instead of in this case `ltp`. And please insert a blank line between line 62 and 63 for readability :wink: